Gordon360/Services/ProfileService.cs

```diff
@@ -301,6 +306,26 @@ public async Task UpdateMobilePrivacyAsync(string username, string value)
             _context.SaveChanges();
         }
 
+        /// <summary>
+        /// privacy setting of mobile phone.
+        /// </summary>
+        /// <param name="username">AD Username</param>
+        /// <param name="value">Y or N</param>
+        /// 
+        /// not being used for now
+        public async Task UpdatePlannedGraduationYearPrivacyAsync(string username, bool value)
+        {
+            var account = _accountService.GetAccountByUsername(username);
+            // Update value in cached data
+            var student = _context.CUSTOM_PROFILE.FirstOrDefault(x => x.username == username);
+            if (student == null)
+            {
+                await _context.CUSTOM_PROFILE.AddAsync(new CUSTOM_PROFILE { username = username, IsPlannedGradYearPrivate = Convert.ToInt32(value) });
+            }
+            student.IsPlannedGradYearPrivate = Convert.ToInt32(value);
```

Comment:
```suggestion
            student.IsPlannedGradYearPrivate = value;
```
once the database has been changed